5.machine_learning/scripts/4.vis_generalizability.r

```diff
@@ -65,6 +65,77 @@ ggsave("./figures/healthy_prob_ridge_plot_plate_3.png", ridge_plot_healthy, heig
 
 ridge_plot_healthy
 
+# Function to create a color palette based on treatments
+create_color_palette <- function(treatments) {
+  colors <- c(
+    "drug_x" = brewer.pal(3, "Dark2")[1],
+    "DMSO" = brewer.pal(3, "Dark2")[2],
+    "TGFRi" = brewer.pal(3, "Dark2")[3]
+  )
+  return(colors[treatments])
+}
+
+# Filter rows for each treatment
+drug_x_wells <- combined_probabilities_df[combined_probabilities_df$Metadata_treatment == "drug_x", ]
+DMSO_wells <- combined_probabilities_df[combined_probabilities_df$Metadata_treatment == "DMSO", ]
+TGFRi_wells <- combined_probabilities_df[combined_probabilities_df$Metadata_treatment == "TGFRi", ]
+
+create_ridge_plot <- function(data, treatment) {
+  # Filter rows where Metadata_cell_type is "Failing"
+  failing_wells <- data[data$Metadata_cell_type == "Failing", ]
+
+  # Create a color palette based on the treatment
+  colors <- create_color_palette(rep(treatment, nrow(failing_wells)))
+  
+  # Ridge plot for average healthy probabilities across wells within each treatment
+  ridge_plot <- ggplot(failing_wells, aes(x = Healthy_probas, y = Metadata_Well)) +
+    geom_density_ridges(aes(fill = Metadata_treatment), alpha = 0.7, scale = 3, rel_min_height = 0.01, bandwidth = 0.1) +  # Change here
+    scale_fill_manual(values = colors) +
+    geom_vline(xintercept = 1, linetype = "dashed", color = "black") +
+    scale_x_continuous(breaks = seq(0, 1, 0.5)) +
+    facet_grid(~model_type, scales = "free_y") +
+    labs(y = "Well") +
+    ggtitle(paste("Average Healthy Probs -", treatment)) +
+  
+    theme_bw() +
+    theme(legend.position = "none",
+          axis.text = element_text(size = 12),
+          axis.text.x = element_text(size = 12),
+          axis.title = element_text(size = 14),
+          strip.text = element_text(size = 14),
+          strip.background = element_rect(
+            colour = "black",
+            fill = "#fdfff4"
+          )
+    )
+  
+  return(ridge_plot)
+}
+
+
+# Create ridge plots for each treatment
+ridge_plot_drug_x <- create_ridge_plot(drug_x_wells, "drug_x")
+ridge_plot_DMSO <- create_ridge_plot(DMSO_wells, "DMSO")
+ridge_plot_TGFRi <- create_ridge_plot(TGFRi_wells, "TGFRi")
+
+# Save figures
+ggsave("./figures/drug_x_healthy_prob_ridge_plot.png", ridge_plot_drug_x, height = 8, width = 12, dpi = 500)
```

Comment:
A practice I like for saving and viewing plots is setting a `width` and `height` variable and passing those values in when saving. This ensures that your saved plot looks like the plot you have been viewing in your IDE.

Reply:
Sounds good! I can definitely try to start implementing that!